Sources/Sentry/SentryInstallation.m

```diff
@@ -1,13 +1,14 @@
 #import "SentryInstallation.h"
 #import "SentryDefines.h"
+#import "SentryOptions.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 @implementation SentryInstallation
 
 static NSString *volatile installationString;
 
-+ (NSString *)id
++ (NSString *)idWithOptions:(SentryOptions *)options
```

Comment:
`h`: If you call this now with different `cacheDirectoryPaths` the result will stick to the first invocation. To avoid having to build the `installationString` on every invocation, we could keep track of these with an NSDictionary.

Reply:
Does this mean that I start the SDK with a custom cache directory, then close the SDK, then start it again with a new cache directory? I would think it's no big deal to rebuild the string on each call to install. Or is there a different scenario you're thinking of?